src/shared/cloudformation/activation.ts

```diff
@@ -79,15 +82,19 @@ function setTemplateRegistryInGlobals(registry: CloudFormationTemplateRegistry)
         set(newInstance: CloudFormationTemplateRegistry) {
             this.cfnInstance = newInstance
         },
-        get() {
+        async get() {
             // This condition handles testing scenarios where we may have
             // already set a mock object before activation.
             // Though in prod nothing should be calling this 'set' function.
             if (this.cfnInstance) {
                 return this.cfnInstance
             }
 
-            return asyncRegistry.getInstance()
+            // prevent eager load if codelenses are off
+            const config = SamCliSettings.instance
+            if (config.get('enableCodeLenses', false) || isToolkitActive()) {
```

Comment:
Ideally we would fix the callers that are actually calling this during startup. But this is a good compromise for now.